lib/users.rb

```diff
@@ -0,0 +1,36 @@
+require 'pg'
+require_relative 'database_connection'
+
+class Users
+  def self.setup
+    @@database = database_connect
+  end
+
+  def self.signup(name, username, email, password)
+    @@database.query("INSERT INTO users (name, username, email, password) VALUES('#{name}', '#{username}', '#{email}', '#{password}');")
+    "#{username} registered"
+  end
+
+  def self.duplicate_email?(email)
+    return false if @@database.query("SELECT email FROM users WHERE email='#{email}'").values == []
+    true
+  end
+
+  def self.duplicate_username?(username)
+    return false if @@database.query("SELECT username FROM users WHERE username='#{username}'").values == []
+    true
+  end
```

Comment:
This is really good, I'm not sure they asked for this but this is a really good feature!